src/uwtools/drivers/forecast.py

```diff
@@ -162,12 +162,28 @@ def job_card(self):
         Turns the resources config object into a batch card for the configured Task.
         """
 
-    def run(self):
-        """
-        Runs the forecast executable with the namelist file and staged input files.
-
-        This will both build the executable and run it.
-        """
+    def run(
+        self,
+        exec_name: str,
+        export: Optional[str] = None,
+        mpi: Optional[str] = None,
+        node_arg: Optional[str] = None,
+        pe_member01: Optional[int] = None,
+        num_of_procs: Optional[int] = None,
+        oversubscribe: Optional[str] = None,
+        procs_per_nodes: Optional[str] = None,
+        ppn: Optional[int] = None,
+        cpu_bind: Optional[str] = None,
+        core: Optional[str] = None,
+        depth: Optional[str] = None,
+        omp_num_threads: Optional[int] = None,
```

Comment:
Since these arguments (all but `self` and `exec_name`) are all optional, can we sort them since their order is irrelevant?

Reply:
Is the order of the arguments always irrelevant in the run command? We weren't sure since all the examples from the various platforms all seemed to follow the same order, which is why we have ordered them in this way here.

Reply:
I'm fairly certain that any `-` or `--` switches are independent of order. Maybe the executable name needs to come last, but I'm not sure even that's the case. Definitely `mpirun` / `mpiexec` / `srun` has to come first.

Speaking of the executable name, did we decide to delegate that to the caller to tack on to the end of the command? I think so...

Reply:
I don't think we'll have cases where the order matters outside the command name (mpirun / mpiexec / srun) first and the executable last. Both should be required arguments. I think it only makes sense to have this method construct the string with the executable name included.